pkg/machinepool/machinepool_test.go

```diff
@@ -225,6 +227,49 @@ var _ = Describe("Machinepool and nodepool", func() {
 				),
 			)
 		})
+		It("should return an error if both `subnet` and `availability-zone` flags are set", func() {
+			cmd := &cobra.Command{}
+			cmd.Flags().Bool("availability-zone", true, "")
+			cmd.Flags().Bool("subnet", true, "")
+
+			clusterKey := "test-cluster-key"
+			clusterBuilder := cmv1.NewCluster().ID("test").State(cmv1.ClusterStateReady)
+			cluster, err := clusterBuilder.Build()
+			Expect(err).ToNot(HaveOccurred())
+			r := &rosa.Runtime{}
+			args := &MachinePoolArgs{}
+
+			cmd.Flags().Set("availability-zone", "true")
+			cmd.Flags().Set("subnet", "true")
+
+			machinePool := &machinePool{}
+			err = machinePool.AddNodePool(cmd, clusterKey, cluster, r, args)
+
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("Setting both `subnet` and " +
+				"`availability-zone` flag is not supported. Please select `subnet` " +
+				"or `availability-zone` to create a single availability zone machine pool"))
+		})
+		It("should fail name validation", func() {
```

Comment:
Same as above, use `When` and `It` to add more details as to why the validation is failing